src/sharedServices/s3/generateS3ObjectPrefixesForDateList.ts

```diff
@@ -1,48 +1,22 @@
-import { getEpochDate } from '../../utils/helpers'
-
 export const generateS3ObjectPrefixesForDateList = (
   dateStrings: string[]
 ): string[] => {
-  const dateEpochs = dateStrings.map(getEpochDate)
+  return dateStrings.map((dateString) => {
+    const dateParts = dateString.split('-')
 
-  const timeRanges: Date[] = dateEpochs.map(generateTimeRangeForDate).flat()
+    validateDate(dateParts)
 
-  return timeRangeToObjectPrefixes(timeRanges)
+    return `firehose/${dateParts[0]}/${dateParts[1]}/${dateParts[2]}`
+  })
 }
 
-const generateTimeRangeForDate = (epochStartDate: number): Date[] => {
-  // Include the last hour of the previous day
-  const iterateHour = new Date(epochStartDate)
-  iterateHour.setUTCHours(iterateHour.getUTCHours() - 1)
-
-  // Set end to be the last hour
-  const lastHour = new Date(epochStartDate)
-  lastHour.setUTCHours(lastHour.getUTCHours() + 23)
-
-  const timeRange: Date[] = []
-
-  while (iterateHour <= new Date(lastHour)) {
-    timeRange.push(new Date(iterateHour))
-    iterateHour.setUTCHours(iterateHour.getUTCHours() + 1)
-  }
-
-  // Include the first hour of the next day
-  timeRange.push(new Date(iterateHour))
-
-  return timeRange
-}
-
-const timeRangeToObjectPrefixes = (timeRange: Date[]): string[] => {
-  return dedupeArray(
-    timeRange.map((date) => {
-      const year = date.getUTCFullYear()
-      const month = (date.getUTCMonth() + 1).toString().padStart(2, '0')
-      const day = date.getUTCDate().toString().padStart(2, '0')
-      const hours = date.getUTCHours().toString().padStart(2, '0')
-
-      return `firehose/${year}/${month}/${day}/${hours}`
-    })
+const validateDate = (dateParts: string[]) => {
+  const epochDate = Date.UTC(
+    parseInt(dateParts[0]),
+    parseInt(dateParts[1]) - 1,
+    parseInt(dateParts[2])
   )
-}
 
-const dedupeArray = (input: string[]) => [...new Set(input)]
+  if (isNaN(epochDate))
```

Comment:
Don't think I see a test for this scenario? I would do that, since you're handling it specifically, as it's always good to keep the coverage up.

Reply:
I have now added tests to cover the input date being a non-date string and a nonsense date string

Reply:
Looks good!